ibis/tests/all/test_client.py

```diff
@@ -38,3 +39,20 @@ def test_query_schema(backend, con, alltypes, expr_fn, expected):
     expected = ibis.schema([(name, dtype(nullable=schema[name].nullable))
                             for name, dtype in expected])
     assert query.schema().equals(expected)
+
+
+@pytest.mark.parametrize('field', [
+    'bool_col', 'tinyint_col', 'smallint_col',
+    'int_col', 'bigint_col', 'float_col', 'double_col',
+    'date_string_col', 'string_col', 'timestamp_col'
+])
+@tu.skipif_unsupported
+def test_sql(backend, con, alltypes, field):
+    try:
+        result = con.sql(alltypes.compile())
+        getattr(result, field)
+        getattr(alltypes, field)
+    except AttributeError:
+        # the method couldn't be set as unsupported but maybe it is not
+        # implemented so it will raise AttributeError.
```

Comment:
Which of these attribute accesses raises the `AttributeError`?

Reply:
Whichever it is, put *only* that line in the `try:` block and move the rest to an `else:` block. Otherwise the `getattr` calls may fail for real reasons and we'll swallow the exception.

Reply:
actually the problem is when some backend doesn't really support `sql`

for example:

`E           AttributeError: 'CSVClient' object has no attribute 'sql'`

or

`E       AttributeError: 'SQLiteClient' object has no attribute '_get_schema_using_query'`

another option would be check if the backend has these two methods first.

Reply:
I will check if the backend has this 2 methods first and if not it will call pytest.skip